databases/src/create/PostgresAccountWizard/createPostgresServer/PostgresServerCreateStep.ts

```diff
@@ -0,0 +1,102 @@
+import { localize } from "../../../utils/localize";
+import { nonNullProp } from "../../../utils/nonNull";
+import { AbstractServerCreate, PostgresServerType } from "../abstract/models";
+import { IPostgresServerWizardContext } from '../IPostgresServerWizardContext';
+
+export class PostgresServerCreateStep extends AzureWizardExecuteStep<IPostgresServerWizardContext> {
+    public priority: number = 150;
+
+    public async execute(context: IPostgresServerWizardContext, progress: Progress<{ message?: string; increment?: number }>): Promise<void> {
+
+        const locationName: string = (await LocationListStep.getLocation(context)).name;
+        const rgName: string = nonNullProp(nonNullProp(context, 'resourceGroup'), 'name');
+        const storageMB: string = nonNullProp(nonNullProp(context, 'sku'), 'size');
+        const newServerName = nonNullProp(context, 'newServerName');
+        const password: string = nonNullProp(context, 'adminPassword');
+
+        return await callWithMaskHandling(
+            async () => {
+                const serverType = nonNullProp(context, 'serverType');
+                const createMessage: string = localize('creatingPostgresServer', 'Creating PostgreSQL Server "{0}"... It should be ready in several minutes.', context.newServerName);
+
+                ext.outputChannel.appendLog(createMessage);
+                progress.report({ message: createMessage });
+                const options: AbstractServerCreate = {
+                    location: locationName,
+                    sku: nonNullProp(context, 'sku'),
+                    administratorLogin: nonNullProp(context, 'shortUserName'),
+                    administratorLoginPassword: password,
+                    storageMB: parseInt(storageMB)
+                };
+
+                switch (serverType) {
+                    case PostgresServerType.Single:
+                        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                        const singleClient = await createPostgreSQLClient(context);
+                        context.server = await singleClient.servers.create(rgName, newServerName, this.asSingleParameters(options));
+                        break;
+                    case PostgresServerType.Flexible:
+                        const flexiClient = await createPostgreSQLFlexibleClient(context);
+                        context.server = await flexiClient.servers.create(rgName, newServerName, this.asFlexibleParameters(options));
+                        break;
+                }
+                context.server.serverType = serverType;
+            },
+            password);
+    }
+
+    public shouldExecute(context: IPostgresServerWizardContext): boolean {
+        return !context.server;
+    }
+
+
+    private asFlexibleParameters(parameters: AbstractServerCreate): FlexibleModels.Server {
+        return {
+            location: parameters.location,
+            version: "12",
```

Comment:
Is this something that may change often? If so maybe extract as a constant.

Reply:
Tracking all suggestions for updating "create"/"utils" folder here: #1013